tests/tls_apis/enc/enc.cpp

```diff
@@ -0,0 +1,179 @@
+    for (int i = 0; i < OE_UNIQUE_ID_SIZE; i++)
+        OE_TRACE_INFO("0x%0x ", (uint8_t)identity->signer_id[i]);
+
+    OE_TRACE_INFO("\nparsed_report->identity.signer_id :\n");
+    for (int i = 0; i < OE_SIGNER_ID_SIZE; i++)
+        OE_TRACE_INFO("0x%0x ", (uint8_t)identity->signer_id[i]);
+
+    OE_TRACE_INFO("\nidentity->product_id :\n");
+    for (int i = 0; i < OE_PRODUCT_ID_SIZE; i++)
+        OE_TRACE_INFO("0x%0x ", (uint8_t)identity->product_id[i]);
+
+    result = OE_OK;
+done:
+    return result;
+}
+
+// input: input_data and input_data_len
+// output: key, key_size
+oe_result_t generate_key_pair(
+    uint8_t** public_key,
+    size_t* public_key_size,
+    uint8_t** private_key,
+    size_t* private_key_size)
+{
+    oe_result_t result = OE_FAILURE;
+    oe_asymmetric_key_params_t params;
+    char user_data[] = "test user data!";
+    size_t user_data_size = sizeof(user_data) - 1;
+
+    OE_TRACE_INFO("Generate key pair");
+
+    params.type = OE_ASYMMETRIC_KEY_EC_SECP256P1; // MBEDTLS_ECP_DP_SECP256R1
+    params.format = OE_ASYMMETRIC_KEY_PEM;
+    params.user_data = user_data;
+    params.user_data_size = user_data_size;
+    result = oe_get_public_key_by_policy(
+        OE_SEAL_POLICY_UNIQUE,
+        &params,
+        public_key,
+        public_key_size,
+        NULL,
+        NULL);
+    OE_CHECK(result);
+
+    result = oe_get_private_key_by_policy(
+        OE_SEAL_POLICY_UNIQUE,
+        &params,
+        private_key,
+        private_key_size,
+        NULL,
+        NULL);
+    OE_CHECK(result);
+
+done:
+    return result;
+}
+
+oe_result_t get_tls_cert(unsigned char** cert, size_t* cert_size)
+{
+    oe_result_t result = OE_FAILURE;
+    uint8_t* host_cert_buf = NULL;
+
+    uint8_t* output_cert = NULL;
+    size_t output_cert_size = 0;
+
+    uint8_t* private_key = NULL;
+    size_t private_key_size = 0;
+    uint8_t* public_key = NULL;
+    size_t public_key_size = 0;
+
+    OE_TRACE_INFO("called into enclave\n");
+
+    // generate public/private key pair
+    result = generate_key_pair(
+        &public_key, &public_key_size, &private_key, &private_key_size);
+    if (result != OE_OK)
+    {
+        OE_TRACE_ERROR(" failed with %s\n", oe_result_str(result));
+        goto done;
+    }
+
+    OE_TRACE_INFO("private key:[%s]\n", private_key);
+    OE_TRACE_INFO("public key:[%s]\n", public_key);
+
+    result = oe_gen_tls_cert(
+        private_key,
+        private_key_size,
+        public_key,
+        public_key_size,
+        &output_cert,
+        &output_cert_size);
+    if (result != OE_OK)
+    {
+        OE_TRACE_ERROR(" failed with %s\n", oe_result_str(result));
+        goto done;
+    }
+
+    OE_TRACE_INFO("output_cert_size = 0x%x", output_cert_size);
+    // validate cert inside the enclave
+    result = oe_verify_tls_cert(
+        output_cert, output_cert_size, enclave_identity_verifier, NULL);
+    OE_TRACE_INFO(
+        "\nFrom inside encalve: verifying SGX certificate extensions... %s\n",
```

Comment:
Delete "SGX", this API should not be SGX specific.  Also "encalve" is misspelled

Reply:
deleted